lucid/overview.py

```diff
@@ -88,6 +152,31 @@ def _show_device_wrapper(self, device):
             partial(self.show_device, device),
             title=device.name)
 
+    def _update_desc(self, device_name: str, device_desc: str) -> None:
+        if self.config_file:
+            logger.debug(f'updating device description: {device_name} = {device_desc}')
+            # open yaml file
+            ryaml = YAML()
+            ryaml.default_flow_style = False
+            if isinstance(self.config_file, (str, bytes, os.PathLike)):
+                fpath = self.config_file
+                with open(self.config_file, 'r') as tf:
+                    config = ryaml.load(tf)
+            else:
+                fpath = os.path.abspath(self.config_file.name)
+                with open(fpath, 'r') as tf:
+                    config = ryaml.load(tf)
+
+            if not config.get('DEVICE_HINTS'):
+                config['DEVICE_HINTS'] = {}
+            if device_desc == config['DEVICE_HINTS'].get(device_name):
+                return
+            if device_desc == '':
+                config['DEVICE_HINTS'].pop(device_name, None)
+            else:
+                config['DEVICE_HINTS'][device_name] = device_desc
+            ryaml.dump(config, Path(fpath))
```

Comment:
Putting it all in the same file is probably the easy way out.  I suppose the case where multiple lucid's are open at the same time will hit upon this rather quickly.

This reminds me of the issue we had in happi where a crash interrupts writing and wipes the file.  I should do the write-new-file -> copy over sequence we implemented there.

Reply:
To me, this feels qualitatively different to `happi` which has a cli tool for editing the file, whereas in this project the user is expected to edit the file themselves.

Reply:
I commented on the typhos PR before making it back to this one... Separating the files makes sense.
Backing up a step, though...

You mention happi, @ZLLentz, did you guys discuss about putting these notes directly into the database? It seems sensible to have shared notes in a database to me. Or is it like different hutches should have different notes about a shared beamline component or something?

Reply:
I think that happi could become a source for notes, but at a lower priority (shadowed by local or env var notes).  This is precisely for the reason you mentioned; different hutches might be looking at the same beamline component and have different notes, and I wouldn't want them to clobber each other.  

The other wrinkle is that happi makes sense for top-level device information, but storing component level notes in happi might get a bit messy (though not completely unreasonable).

Reply:
I feel like we've talked about storing extra data in happi as a go-to per-device data source a bunch of times in many contexts, which makes me think that it's worth doing. It might even be worth storing subdevice notes along with the device notes, I'm not sure? It's not immediately clear to me and warrants some discussion and mock-ups.

Reply:
That's a lot of sources of notes and documentation... Off the top of my head, correct me if I'm wrong:
1. LUCID config file 1 - env var notes
2. LUCID config file 2 - local notes
3. typhos config file 1 - env var notes
4. typhos config file 2 - local notes
5. happi database
6. confluence - PCDS
7. confluence - mechanical team / hutch team / etc
8. FRS / ESD / ...
9. Design review slides...

Reply:
To some extent the Typhos PR could supercede this one, or at least LUCID/Typhos could use the same files.  (so maybe that simplifies things slightly)

Adding yet another source of documentation that can fall out of date leaves a slightly bitter after taste, but I could argue it fills a slightly different niche